core/arch/arm/mm/core_mmu_lpae.c

```diff
@@ -530,6 +588,94 @@ static void core_init_mmu_prtn_ta_va_find(struct mmu_partition *prtn)
 	assert(user_va_idx != -1);
 }
 
+/*
+ * TAs page table is an entry inside level 1 page table.
+ *
+ * If base table level is 1 - user_va_idx is already the index,
+ *                            so nothing to do.
+ * If base table level is 0 - we might need to allocate entry 0 of base table,
+ *                            as TAs page table is an entry inside a level 1
+ *                            page table.
+ */
+static void core_init_mmu_prtn_ta_core(struct mmu_partition *prtn
+				       __maybe_unused,
+				       unsigned int base_idx __maybe_unused,
+				       unsigned int core __maybe_unused)
+{
+#if (CORE_MMU_BASE_TABLE_LEVEL == 0)
+	uint64_t *tbl = NULL;
+	uintptr_t idx = 0;
+
+	assert(user_va_idx != -1);
+	COMPILE_TIME_ASSERT(MAX_XLAT_TABLES <
+			    (1 << (8 * sizeof(prtn->user_l1_table_idx[0][0]))));
+
+	tbl = prtn->base_tables[base_idx][core];
+
+	/*
+	 * If base level is 0, then user_va_idx refers to
+	 * level 1 page table that's in base level 0 entry 0.
+	 */
+	if (CORE_MMU_BASE_TABLE_LEVEL == 0) {
+		struct core_mmu_table_info tbl_info = { };
+		bool ret = false;
+
+		core_mmu_set_info_table(&tbl_info, 0, 0, tbl);
+#ifdef CFG_VIRTUALIZATION
+		tbl_info.prtn = prtn;
+#endif
+
+		/*
+		 * If this isn't the core that created the initial tables
+		 * mappings, then the level 1 table must be copied,
+		 * as it will hold pointer to the user mapping table
+		 * that changes per core.
+		 */
+		if (core != get_core_pos()) {
+			ret = core_mmu_entry_copy(&tbl_info, 0);
+			assert(ret);
+		}
+
+		ret = core_mmu_entry_to_finer_grained(&tbl_info, 0, true);
+		assert(ret);
+
+		/*
+		 * Now base level table should be ready with a table descriptor
+		 */
+		assert((tbl[0] & DESC_ENTRY_TYPE_MASK) == TABLE_DESC);
+
+		tbl = core_mmu_xlat_table_entry_pa2va(prtn, 0, tbl[0]);
+
+		assert(tbl);
+	}
+
+	idx = ((uintptr_t)&tbl[user_va_idx] -
```

Comment:
(nitpicking) would prefer:
```c
 	idx = ((uintptr_t)&tbl[user_va_idx] - (uintptr_t)prtn->xlat_tables) /
	      sizeof(xlat_tbl_t);
```

Reply:
Fixed, but it will likely to fail checkpatch...
```
$ checkpatch.pl -g HEAD~
ERROR: space prohibited after that '-' (ctx:WxW)
#229: FILE: core/arch/arm/mm/core_mmu_lpae.c:649:
+       idx = ((uintptr_t)&tbl[user_va_idx] - (uintptr_t)prtn->xlat_tables) /
                                            ^

total: 1 errors, 0 warnings, 0 checks, 396 lines checked
```